### PR TITLE
feat: Add ability to specify the records id when calling model.create()

### PR DESCRIPTION
**Pull request by jamemackson**
@zacharygolba here's the first pass, just pushing to start the discussion on the implementation of this.

refs: #678

### CODE REVIEW

**Comment by jamemackson** [comment]
@zacharygolba I updated this to use a property on the model that enables the alternate behavior to allow, on a model specific basis,  inserting with a specified id.  I'm sure there are still a lot of ways this could be better... I'm all ears for any suggestions you have.

File: src/packages/database/model/index.js [hunk 1]
```diff
@@ -592,6 +592,16 @@ class Model {
...
 
   /**
+   * allows the internal hooks to define the id value
+   *
+   * @property allowPrimaryKeyInsert
+   * @type {boolean}
+   * @default 'id'
+   * @public
+   */
+  allowPrimaryKeyInsert: boolean = false;
```
Comment (zacharygolba) [inline src/packages/database/model/index.js]: I think that this flag is unnecessary. The ORM should be smart enough to know if a record has an id in it's change set and it has not been persisted to the database yet.

File: src/packages/database/model/utils/persistence.js [hunk 2]
```diff
@@ -23,6 +23,24 @@ export function create(record: Model, trx: Object): Array<Object> {
     updatedAt: timestamp
   });
 
+  if (record.allowPrimaryKeyInsert) {
+    const columns = getColumns(record);
+    // check that we have a valid id here?
```
Comment (zacharygolba) [inline src/packages/database/model/utils/persistence.js]: I don't think that we have to check if the id is valid. If this fails due to an error that is thrown by the database driver that is ok. No need for us to add an additional error message and spend more time in node than what is necessary.

File: src/packages/database/test/model.test.js [hunk 3]
```diff
@@ -382,6 +382,35 @@ describe('module "database/model"', () => {
...
+
+        result = await Subject.create({
+          id,
+          user,
+          body,
+          title,
+          isPublic: true
+        });
+
+        expect(result).to.be.an.instanceof(Subject);
```
Comment (zacharygolba) [inline src/packages/database/test/model.test.js]: I feel like this assertion and `expect(result).to.have.property('id', id);` from L403 are all that is required to ensure that you can manually insert ids. Maybe w could remove all of the other expect calls and the unnecessary constants for this test case.

**Comment by jamemackson** [comment]
@zacharygolba thanks for the feedback, all great suggestions.  I hope the modifications to the test was inline with what you were thinking.  Let me know... :)

**Comment by zacharygolba** [comment]
Looks good @jamemackson. Awesome work!

**Comment by jamemackson** [comment]
thanks @zacharygolba!!

**Pull request merged**